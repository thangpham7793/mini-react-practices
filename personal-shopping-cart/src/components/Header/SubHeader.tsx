import React from "react";

export const SubHeader = () => {
  return <div></div>;
};
