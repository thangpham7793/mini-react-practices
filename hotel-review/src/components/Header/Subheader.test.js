import React from "react";
import { shallow } from "enzyme";
import SubHeader, { Title, SubHeaderButton } from "./SubHeader";

describe("Subheader", () => {
  it("should render", () => {
    const component = shallow(<SubHeader />);
    expect(component).toMatchSnapshot();
  });

  it("should render with a dynamic title", () => {
    const title = "Title";
    const component = shallow(<SubHeader title={title} />);
    expect(component.find(Title).text()).toEqual(title);
  });

  it("should render a goBack & openForm button that handle click events", () => {
    const mockGoBack = jest.fn();
    const mockOpenForm = jest.fn();
    const component = shallow(
      <SubHeader goBack={mockGoBack} openForm={mockOpenForm} />
    );
    const goBackButton = component.find(SubHeaderButton).at(0);
    const openFormButton = component.find(SubHeaderButton).at(1);

    expect(openFormButton.exists()).toBe(true);
    expect(goBackButton.exists()).toBe(true);

    goBackButton.simulate("click");
    expect(mockGoBack).toHaveBeenCalled();

    openFormButton.simulate("click");
    expect(mockOpenForm).toHaveBeenCalled();
  });
});
