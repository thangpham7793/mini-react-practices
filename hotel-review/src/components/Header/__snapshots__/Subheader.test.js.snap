// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subheader should render 1`] = `ShallowWrapper {}`;
